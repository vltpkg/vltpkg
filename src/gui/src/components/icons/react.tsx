export const React = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      width="500"
      height="500"
      viewBox="0 0 500 500"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}>
      <g clip-path="url(#clip0_2967_809)">
        <path
          d="m250.504 284.862c270.046 284.862 285.889 269.029 285.889 249.499c285.889 229.968 270.046 214.136 250.504 214.136c230.961 214.136 215.119 229.968 215.119 249.499c215.119 269.029 230.961 284.862 250.504 284.862z"
          fill="currentColor"
        />
        <path
          d="m250.501 321.96c355.363 321.96 440.371 289.522 440.371 249.509c440.371 209.495 355.363 177.058 250.501 177.058c145.639 177.058 60.6318 209.495 60.6318 249.509c60.6318 289.522 145.639 321.96 250.501 321.96z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.724 285.725c240.155 376.482 310.768 433.837 345.442 413.83c380.116 393.823 365.721 304.031 313.29 213.274c260.859 122.517 190.247 65.1621 155.572 85.1689c120.898 105.176 135.293 194.968 187.724 285.725z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.726 213.277c135.295 304.034 120.9 393.826 155.574 413.833c190.248 433.84 260.861 376.485 313.292 285.728c365.723 194.971 380.118 105.179 345.443 85.172c310.769 65.1652 240.157 122.52 187.726 213.277z"
          stroke="currentColor"
          stroke-width="18.913"
        />
      </g>
      <defs>
        <clipPath id="clip0_2967_809">
          <rect
            width="397"
            height="353"
            fill="currentColor"
            transform="translate(52 73)"
          />
        </clipPath>
      </defs>
    </svg>
  )
}
