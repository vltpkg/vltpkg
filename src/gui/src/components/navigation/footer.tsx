import ThemeSwitcher from '@/components/theme-switcher/theme-switcher.jsx'
import {
  Discord,
  Linkedin,
  Github,
  TwitterX,
} from '@/components/icons/index.js'

interface SocialMediaLink {
  name: string
  to: string
  component: React.ComponentType<React.SVGProps<SVGSVGElement>>
}

const socialMediaLinks: SocialMediaLink[] = [
  {
    name: 'linkedin',
    to: 'https://www.linkedin.com/company/vltpkg/',
    component: props => <Linkedin {...props} />,
  },
  {
    name: 'twitter-x',
    to: 'https://x.com/vltpkg',
    component: props => <TwitterX {...props} />,
  },
  {
    name: 'github',
    to: 'https://github.com/vltpkg',
    component: props => <Github {...props} />,
  },
  {
    name: 'discord',
    to: 'https://discord.gg/vltpkg',
    component: props => <Discord {...props} />,
  },
]

export const Footer = () => {
  return (
    <footer>
      <div className="flex w-full flex-col gap-x-4 gap-y-4 px-6 py-6">
        {/* footer links */}
        <div className="flex w-full items-center justify-between">
          <div className="flex gap-4">
            {socialMediaLinks.map((link, idx) => (
              <a
                href={link.to}
                key={idx}
                aria-label={link.name}
                target="_blank">
                <link.component className="h-5 fill-black dark:fill-white" />
              </a>
            ))}
          </div>
          <div className="flex">
            <ThemeSwitcher />
          </div>
        </div>

        {/* footer policies */}
        <div className="flex w-full flex-row items-center justify-between">
          <a
            href="https://www.vlt.sh/"
            className="text-sm text-muted-foreground no-underline transition-all hover:text-foreground">
            &copy; {new Date().getFullYear()} vlt technology inc.
          </a>
          <div className="flex flex-row gap-4">
            <a
              href="https://www.vlt.sh/terms"
              className="text-sm text-muted-foreground no-underline transition-all hover:text-foreground">
              Terms
            </a>
            <a
              href="https://www.vlt.sh/privacy"
              className="text-sm text-muted-foreground no-underline transition-all hover:text-foreground">
              Privacy
            </a>
          </div>
        </div>
      </div>
    </footer>
  )
}
