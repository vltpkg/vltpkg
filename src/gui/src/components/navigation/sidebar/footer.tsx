import {
  SidebarMenuItem,
  SidebarMenuButton,
  SidebarMenu,
  SidebarFooter,
  SidebarSeparator,
} from '@/components/ui/sidebar.jsx'
import { Library, ArrowUpRight } from 'lucide-react'
import { SidebarTrigger } from '@/components/navigation/sidebar/trigger.jsx'

const AppSidebarFooter = () => {
  return (
    <SidebarFooter>
      <SidebarMenu>
        <SidebarMenuItem>
          <SidebarMenuButton asChild>
            <a href="https://docs.vlt.sh" target="_blank">
              <Library />
              <span>Documentation</span>
              <ArrowUpRight className="ml-auto" />
            </a>
          </SidebarMenuButton>
        </SidebarMenuItem>
      </SidebarMenu>
      <SidebarSeparator />
      <SidebarTrigger />
    </SidebarFooter>
  )
}

export { AppSidebarFooter }
