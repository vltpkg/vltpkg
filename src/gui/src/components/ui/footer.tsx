import { useTheme } from './theme-provider.jsx'

interface SocialMediaLink {
  name: string
  to: string
}

const socialMediaLinks: SocialMediaLink[] = [
  {
    name: 'linkedin',
    to: 'https://www.linkedin.com/company/vltpkg/',
  },
  {
    name: 'twitter-x',
    to: 'https://x.com/vltpkg',
  },
  {
    name: 'github',
    to: 'https://github.com/vltpkg',
  },
  {
    name: 'discord',
    to: 'https://discord.gg/vltpkg',
  },
]

export const Footer = () => {
  const { theme } = useTheme()

  return (
    <footer className="flex border-t-[1px] border-solid h-24 items-center justify-between px-8 w-full">
      <div className="flex items-center gap-3">
        {socialMediaLinks.map((link, idx) => (
          <a href={link.to} key={idx}>
            <img
              src={`/icons/${link.name}.svg`}
              width={20}
              style={{
                filter: theme === 'dark' ? 'invert(0)' : 'invert(1)',
              }}
            />
          </a>
        ))}
      </div>
      <div className="flex items-center">
        <p className="text-md text-muted-foreground">
          &copy; {new Date().getFullYear()} vlt technology inc.
        </p>
      </div>
      <div />
    </footer>
  )
}
