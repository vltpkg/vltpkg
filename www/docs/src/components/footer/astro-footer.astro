---
import type { Props } from '@astrojs/starlight/props'
import Pagination from '@/components/pagination/astro-pagination.astro'
---

<Pagination {...Astro.props} />
