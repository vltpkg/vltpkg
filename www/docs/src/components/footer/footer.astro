---
import type { Props } from '@astrojs/starlight/props'
import Pagination from '@/components/pagination/pagination.astro'
---

<Pagination {...Astro.props} />
