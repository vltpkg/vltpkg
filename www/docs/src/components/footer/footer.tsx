import { type Props } from '@astrojs/starlight/props'
import config from 'virtual:starlight/user-config'
import ThemeSelect from '@/components/theme-select/theme-select'
import { useStore } from '@/state'

const Footer = (_props: Props) => {
  return (
    <footer className="border-t-[1px]">
      {/* footer links */}
      <div className="mx-auto flex w-full max-w-screen-xl flex-col gap-x-4 gap-y-4 px-6 py-6">
        <div className="flex w-full flex-row items-center justify-between">
          <FooterSocials />
          <div className="flex items-center gap-2">
            <a
              href="https://www.vlt.sh/join"
              className="group inline-flex items-center gap-3 rounded-[12px] pl-3 pr-1 text-foreground no-underline transition-all">
              <span className="text-15 rounded-[7px] border border-neutral-300 bg-gradient-to-b from-neutral-50 to-neutral-100 px-3 py-1 text-left font-medium group-hover:border-neutral-400 dark:border-neutral-800 dark:from-neutral-900 dark:to-black dark:group-hover:border-neutral-700 dark:group-hover:from-neutral-800/80">
                Join waitlist
              </span>
            </a>
            <ThemeSelect />
          </div>
        </div>

        {/* footer policies */}
        <div className="flex w-full flex-row items-center justify-between">
          <a
            href="https://www.vlt.sh/"
            className="text-sm font-medium text-muted-foreground no-underline transition-all hover:text-foreground">
            &copy; {new Date().getFullYear()} vlt technology inc.
          </a>
          <div className="flex flex-row gap-4">
            <a
              href="https://www.vlt.sh/terms"
              className="text-sm font-medium text-muted-foreground no-underline transition-all hover:text-foreground">
              Terms
            </a>
            <a
              href="https://www.vlt.sh/privacy"
              className="text-sm font-medium text-muted-foreground no-underline transition-all hover:text-foreground">
              Privacy
            </a>
          </div>
        </div>
      </div>
    </footer>
  )
}

const FooterSocials = () => {
  const { getResolvedTheme } = useStore()
  const theme = getResolvedTheme()

  const socialLinks = Object.entries(config.social ?? {}).map(
    ([platform, value]) => ({
      platform,
      ...value,
    }),
  )

  return (
    <div className="flex gap-4">
      {socialLinks.map((link, idx) => (
        <a href={link.url} key={idx}>
          <img
            src={`/icons/${link.platform}.svg`}
            className="h-5"
            style={{
              filter: theme === 'dark' ? 'invert(0)' : 'invert(1)',
            }}
          />
        </a>
      ))}
    </div>
  )
}

export default Footer
