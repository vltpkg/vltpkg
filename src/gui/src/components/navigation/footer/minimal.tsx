import { useLocation } from 'react-router'
import { ThemeSwitcher } from '@/components/ui/theme-switcher.tsx'
import { useSearchResultsStore } from '@/state/search-results.ts'
import { SearchResultsPaginationNavigation } from '@/components/search/search-results/page-navigation.tsx'
import { SearchResultPageOptions } from '@/components/search/search-results/page-options.tsx'
import { cn } from '@/lib/utils.ts'

import { Fragment } from 'react'
import type { ComponentProps } from 'react'

type MinimalFooterProps = ComponentProps<'footer'>

export const MinimalFooter = ({
  className,
  ...rest
}: MinimalFooterProps) => {
  const { pathname } = useLocation()
  const total = useSearchResultsStore(state => state.total)
  const isLoading = useSearchResultsStore(state => state.isLoading)
  const isSearchRoute = pathname === '/search'

  const displaySearchPagination =
    isSearchRoute && total !== 0 && !isLoading

  return (
    <footer
      className={cn(
        'flex flex-col items-start gap-4 px-4 py-6 md:grid md:grid-cols-12 md:items-center',
        className,
      )}
      {...rest}>
      {displaySearchPagination && (
        <Fragment>
          <SearchResultPageOptions className="col-span-3" />
          <SearchResultsPaginationNavigation className="col-span-6 col-start-4" />
        </Fragment>
      )}
      <div
        className={cn(
          'col-span-3 flex items-center justify-end',
          !displaySearchPagination && 'col-span-full',
        )}>
        <ThemeSwitcher />
      </div>
    </footer>
  )
}
