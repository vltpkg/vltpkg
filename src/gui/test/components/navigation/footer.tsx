import { afterEach, it, describe, vi, expect } from 'vitest'
import { cleanup, render, screen } from '@testing-library/react'
import html from 'diffable-html'
import { useGraphStore as useStore } from '@/state/index.js'
import { Footer } from '@/components/navigation/footer.jsx'

vi.mock('useTheme', () => ({
  useTheme: vi.fn(),
}))
vi.mock('useGraphStore', () => ({
  useGraphStore: vi.fn(),
}))

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

afterEach(() => {
  const CleanUp = () => (useStore(state => state.reset)(), '')
  render(<CleanUp />)
  cleanup()
})

const socialLinks = ['linkedin', 'twitter-x', 'github', 'discord']

describe('Footer Component', () => {
  it('renders the social media links for light theme', () => {
    const Container = () => {
      const setTheme = useStore(state => state.updateTheme)
      setTheme('light')
      return <Footer />
    }
    const { container } = render(<Container />)

    const icons = container.getElementsByTagName('img')
    socialLinks.forEach((link, idx) => {
      const icon = icons[idx] as HTMLImageElement
      expect(icon).toBeDefined()
      expect(icon.src).toContain(`/icons/${link}.svg`)
      const iconStyle = window.getComputedStyle(icon)
      expect(iconStyle.filter).toBe('invert(1)')
    })

    expect(container.innerHTML).toMatchSnapshot()
  })

  it('renders the social media links for dark theme', () => {
    const Container = () => {
      const setTheme = useStore(state => state.updateTheme)
      setTheme('dark')
      return <Footer />
    }
    const { container } = render(<Container />)

    const icons = container.getElementsByTagName('img')
    socialLinks.forEach((link, idx) => {
      const icon = icons[idx] as HTMLImageElement
      expect(icon).toBeDefined()
      expect(icon.src).toContain(`/icons/${link}.svg`)
      const iconStyle = window.getComputedStyle(icon)
      expect(iconStyle.filter).toBe('invert(0)')
    })
    expect(container.innerHTML).toMatchSnapshot()
  })

  it('renders the copyright text', () => {
    const Container = () => {
      return <Footer />
    }

    const { container } = render(<Container />)

    const currentYear = new Date().getFullYear()
    const copyrightText = screen.getByText(
      `Â© ${currentYear} vlt technology inc.`,
    )
    expect(copyrightText).toBeDefined()
    expect(container.innerHTML).toMatchSnapshot()
  })
})
