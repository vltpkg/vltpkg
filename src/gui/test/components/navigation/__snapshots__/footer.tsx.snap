// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > renders the copyright text 1`] = `

<footer class="border-t-[1px] bg-white dark:bg-black">
  <div class="flex w-full flex-col gap-x-4 gap-y-4 px-6 py-6">
    <div class="flex w-full items-center justify-between">
      <div class="flex gap-4">
        <a
          href="https://www.linkedin.com/company/vltpkg/"
          aria-label="linkedin"
        >
          <gui-linkedin-icon classname="h-5 fill-black dark:fill-white">
          </gui-linkedin-icon>
        </a>
        <a
          href="https://x.com/vltpkg"
          aria-label="twitter-x"
        >
          <gui-twitter-icon classname="h-5 fill-black dark:fill-white">
          </gui-twitter-icon>
        </a>
        <a
          href="https://github.com/vltpkg"
          aria-label="github"
        >
          <gui-github-icon classname="h-5 fill-black dark:fill-white">
          </gui-github-icon>
        </a>
        <a
          href="https://discord.gg/vltpkg"
          aria-label="discord"
        >
          <gui-discord-icon classname="h-5 fill-black dark:fill-white">
          </gui-discord-icon>
        </a>
      </div>
      <div class="flex">
        <gui-theme-switcher>
        </gui-theme-switcher>
      </div>
    </div>
    <div class="flex w-full flex-row items-center justify-between">
      <a
        href="https://www.vlt.sh/"
        class="text-sm text-muted-foreground no-underline transition-all hover:text-foreground"
      >
        Â© 2025 vlt technology inc.
      </a>
      <div class="flex flex-row gap-4">
        <a
          href="https://www.vlt.sh/terms"
          class="text-sm text-muted-foreground no-underline transition-all hover:text-foreground"
        >
          Terms
        </a>
        <a
          href="https://www.vlt.sh/privacy"
          class="text-sm text-muted-foreground no-underline transition-all hover:text-foreground"
        >
          Privacy
        </a>
      </div>
    </div>
  </div>
</footer>

`;
